File: src/com/designpatterns/creational/factory/Rectangle.java
package com.designpatterns.creational.factory;

public class Rectangle implements IShape {

    public static final String TAG = "Rectangle";

    private final float length;
    private final float width;

    public Rectangle(float length, float width) {
        this.length = length;
        this.width = width;
    }

    @Override
    public float area() {
        return length * width;
    }

    @Override
    public float perimeter() {
        return 2 * (length + width);
    }

    @Override
    public void draw() {
        System.out.printf("Rectangle:: area:%f, perimeter:%f%n", area(), perimeter());
    }
}

File: src/com/designpatterns/creational/abstractfactory/kotlin/Rectangle.kt
package com.designpatterns.creational.abstractfactory.kotlin

class Rectangle(private val length: Float, private val width: Float) : IShape {

    override fun area(): Float {
        return length * width
    }

    override fun perimeter(): Float {
        return 2 * (length + width)
    }

    override fun draw() {
        System.out.printf("Rectangle:: area:%f, perimeter:%f%n", area(), perimeter())
    }

    companion object {
        const val TAG = "Rectangle"
    }
}